src/app/pages/EarnTez/components/baker-card.tsx

```diff
@@ -0,0 +1,165 @@
+    baker: bakerOrAddress,
+    onClick
+  }: BakerCardProps) => {
+    const { rpcBaseURL, chainId } = network;
+
+    const [hovered, setHovered, setUnhovered] = useBooleanState(false);
+    const { data: stakedData, mutate } = useStakedAmount(rpcBaseURL, accountPkh);
+    const bakerPkh = getBakerAddress(bakerOrAddress);
+    const bakerFromProps = typeof bakerOrAddress === 'object' ? bakerOrAddress : undefined;
+    const explorerHref = useBlockExplorerHref(TempleChainKind.Tezos, chainId, 'address', bakerPkh);
+    const { data: baker = bakerFromProps } = useKnownBaker(
+      bakerFromProps ? null : bakerPkh,
+      network.chainId,
+      !bakerFromProps
+    );
+
+    useOnTezosBlock(rpcBaseURL, () => void mutate());
+
+    const { symbol } = getTezosGasMetadata(chainId);
+
+    const stakedAtomic = useMemo(() => stakedData?.toNumber() || 0, [stakedData]);
+
+    const headerRight = useMemo(
+      () => HeaderRight && <HeaderRight staked={stakedAtomic} />,
+      [HeaderRight, stakedAtomic]
+    );
+
+    const bakerAvatar = <BakerAvatar className="mr-2" address={baker?.address} bakerName={baker?.name} />;
+    const bakerName = <BakerName>{baker ? baker.name : <T id="unknownBakerTitle" />}</BakerName>;
+
+    const delegationFeePercentage = useMemo(() => (baker ? toPercentage(baker.delegation.fee) : '-'), [baker]);
+    const stakingFeePercentage = useMemo(() => (baker ? toPercentage(baker.staking.fee) : '-'), [baker]);
+    const stakingApyPercentage = useMemo(() => (baker ? toPercentage(baker.staking.estimatedApy) : '-'), [baker]);
+
+    const handleClick = useCallback(() => {
+      onClick?.(baker || bakerOrAddress);
+    }, [onClick, baker, bakerOrAddress]);
+
+    const onLinkClick = useCallback<MouseEventHandler<HTMLAnchorElement>>(e => void e.stopPropagation(), []);
+
+    return (
+      <StakingCard
+        className={clsx(onClick && 'cursor-pointer', className)}
+        onClick={handleClick}
+        topInfo={
+          <>
+            {explorerHref ? (
+              <Anchor
+                href={explorerHref}
+                className={clsx('flex items-center', hovered && 'text-secondary')}
+                onClick={onLinkClick}
+                onMouseEnter={setHovered}
+                onMouseLeave={setUnhovered}
+              >
+                {bakerAvatar}
+                {bakerName}
+                <IconBase size={12} Icon={OutLinkIcon} className={clsx('ml-0.5', !hovered && 'invisible')} />
+              </Anchor>
+            ) : (
+              <div className="flex items-center">
+                {bakerAvatar}
+                {bakerName}
+              </div>
+            )}
+            {headerRight}
+          </>
+        }
+        bottomInfo={
+          metricsType === 'delegation' ? (
+            <>
+              <BakerStatsEntry
+                name={t('delegated')}
+                value={baker ? toShortened(baker.delegation.capacity - baker.delegation.freeSpace) : '-'}
```

Comment:
```suggestion
                value={toShortened(baker.delegation.capacity - baker.delegation.freeSpace, '-')}
```

add fallback handling for this fn too